Clinical-Genomics/scout
# Update genes from checked, downloaded or pre-downloaded resource files

Improve the functionality of the `scout update genes` command by making it possible to use pre-downloaded resource files. Allow a **better control over the required files** and the command will **stop without dropping the existing genes** if pre-downloaded files are missing or don't contain valid info.

Fix #2691 and fix #2729

**How to test: local instance of Scout**:
1. Load Scout genes from pre-downloaded files. Download all necessary files by running `scout --demo download everything (--api-key key)`
2. Update genes in one and/or both builds by running `scout --demo update genes -f resources_download_path (--build buildn)`
3. Make sure that the option to download and parse all files on the fly still works as it should:
`scout --demo update genes (--api-key key --build builn)`

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
